Server/modules/luaModules/src/CLuaModulesModule.cpp

```diff
@@ -0,0 +1,104 @@
+#include "StdInc.h"
+#include <sdk/SharedUtil.File.h>
+
+void CLuaModulesModule::Load()
+{
+    m_pLuaModuleManager = new CLuaModuleManager(m_pModuleInterface);
+    
+}
+
+void CLuaModulesModule::UnLoad()
+{
+}
+
+void CLuaModulesModule::DoPulse()
+{
+    m_pLuaModuleManager->DoPulse();
+}
+
+void CLuaModulesModule::ResourceStarted(IResource* pResource)
+{
+}
+
+void CLuaModulesModule::ResourceStopped(IResource* pResource)
+{
+    m_pLuaModuleManager->ResourceStopped(pResource);
+}
+
+void CLuaModulesModule::ResourceStopping(IResource* pResource)
+{
+    m_pLuaModuleManager->ResourceStopping(pResource);
+}
+
+int CLuaModulesModule::LoadModule(const char* szShortFileName, const char* szFileName, bool bLateLoad)
+{
+    return m_pLuaModuleManager->LoadModule(szShortFileName, szFileName, bLateLoad);
+}
+
+int CLuaModulesModule::UnloadModule(const char* szShortFileName)
+{
+    return m_pLuaModuleManager->UnloadModule(szShortFileName);
+}
+
+int CLuaModulesModule::ReloadModule(const char* szShortFileName, const char* szFileName, bool bLateLoad)
+{
+    return m_pLuaModuleManager->ReloadModule(szShortFileName, szFileName, bLateLoad);
+}
+
+ILuaModule* CLuaModulesModule::GetModuleByName(const char* szName)
+{
+    std::vector<CLuaModule*> luaModules = m_pLuaModuleManager->GetLoadedModules();
+    for (const auto luaModule : luaModules)
+    {
+        if (luaModule->_GetName() == szName)
+        {
+            return luaModule;
+        }
+    }
+    return nullptr;
+}
+
+std::vector<ILuaModule*> CLuaModulesModule::GetModules()
+{
+    std::vector<CLuaModule*> luaModules = m_pLuaModuleManager->GetLoadedModules();
+    std::vector<ILuaModule*> modules(luaModules.size());
+    for (const auto luaModule : luaModules)
+    {
+        modules.push_back(luaModule);
+    }
+    return modules;
+}
+
+bool CLuaModulesModule::IsModuleRunning(std::string moduleName)
+{
+    std::vector<CLuaModule*> luaModules = m_pLuaModuleManager->GetLoadedModules();
+    for (const auto luaModule : luaModules)
+    {
+        if (luaModule->_GetName() == moduleName)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+std::vector<SFoundModule> CLuaModulesModule::GetAllModules()
+{
+    std::vector<SFoundModule> foundModules;
+    std::vector<SString>      itemList = SharedUtil::FindFiles(SharedUtil::PathJoin(m_pModuleInterface->GetModulesPath(), "*"), true, false);
+    for (auto const& file : itemList)
+    {
+        SFoundModule foundModule;
+        foundModule.name = file;
+        if (IsModuleRunning(file))
+        {
+            foundModule.state = "running";
+        }
+        else
+        {
+            foundModule.state = "loaded";
+        }
+        foundModules.push_back(foundModule);
```

Comment:
You really should be using Enums for the state, instead of strings.
Also, move semantics would be nice:
```cpp

// Using this instead of `SFoundModule`. Could also just use an std::pair to be honest...
struct ModuleDescriptor {
    std::string path;
    enum class State {
        LOADED,
        RUNNING,
    } state;
}


std::vector<SString> itemList = SharedUtil::FindFiles(SharedUtil::PathJoin(m_pModuleInterface->GetModulesPath(), "*"), true, false);

std::vector<SFoundModule> foundModules;
foundModules.reserve(itemList.size());

for (auto& file : itemList)
{
    foundModules.emplace_back(
        std::move(file),
        IsModuleRunning(file) ? ModuleDescriptor::State::RUNNING : ModuleDescriptor::State::LOADED
    );
}
```

Tbh, the naming of the function is kinda off... You have `GetModules`, which also returns all modules, from what I see.
Could you please add comments to the code explaining what each function does?
If necessary, please rename the function as well. To me it seems like its more like `FindAllModules` or something.